### PR TITLE
Use central differencing in solenoidal correction and tendency calculation

### PR DESCRIPTION
**Pull request by matt-frey**
This PR
- use FD for performing vertical derivatives
- fixes the subroutine diffz
- adds more unit tests for diffz and central_diffz

### CODE REVIEW

File: src/3d/inversion/inversion.f90 [hunk 1]
```diff
@@ -342,23 +344,18 @@ subroutine divergence(f, div)
             ! calculate df1/dx
             call fftxyp2s(f(0:nz, :, :, I_X), fs)
             call diffx(fs, ds)
-            call fftxys2p(ds, div)
+            call fftxys2p(ds, f(0:nz, :, :, I_X))
```
Comment (daviddritschel) [inline src/3d/inversion/inversion.f90:347]: Is it ok that we overwrite f here; is it not needed in the calling routine?

File: src/3d/inversion/inversion.f90 [hunk 2]
```diff
@@ -342,23 +344,18 @@ subroutine divergence(f, div)
...
 
             ! calculate df2/dy
             call fftxyp2s(f(0:nz, :, :, I_Y), fs)
             call diffy(fs, ds)
             call fftxys2p(ds, f(0:nz, :, :, I_Y))
```
Comment (daviddritschel) [inline src/3d/inversion/inversion.f90:352]: Same here

**Review by daviddritschel — COMMENTED** [review]
Just a couple of minor questions - can commit if they are not a concern

File: src/3d/inversion/inversion.f90 (hunk 1 above)
Comment (matt-frey) [reply src/3d/inversion/inversion.f90:347]: It is fine. The call to `fftxyp2s` which convertes f to semi-spectral spce (stored in `fs`) also overwrites the array `f`.

Comment (matt-frey) [reply src/3d/inversion/inversion.f90:347]: The "divergence" routine also has a comment saying "Note: f is overwritten".

File: src/3d/inversion/inversion.f90 (hunk 2 above)
Comment (matt-frey) [reply src/3d/inversion/inversion.f90:352]: See my other reply.

**Pull request merged**